kvserver,kv: move bulk, cloud, and replace kv/*.go

tbg opened the pull request:
`pkg/kvserver/bulk` will move to `pkg/kv/bulk`, `pkg/kvserver/cloud` becomes `pkg/storage/cloud`, and `pkg/kv/*.go` becomes the `pkg/kv/kvclient/kvcoord` package. `pkg/kv` becomes filed with the contents of `pkg/internal/client`.

100% mechanical.

tbg committed:
bulk: git mv to ./pkg/kv

Pure code movement, does not compile.

Release note: None

tbg committed:
bulk: Goland-move to ./pkg/kv/bulk

Release note: None

tbg committed:
cloud: git mv to ./pkg/storage/cloud

Release note: None

tbg committed:
cloud: Goland-move to ./pkg/storage/cloud

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45873)
<!-- Reviewable:end -->

tbg committed:
kv: git mv move package content to kvclient/kvcoord

Release note: None

tbg committed:
kv: Goland-move to kvclient/kvcoord

Release note: None

tbg committed:
kv,kvclient: add placeholder files

TestForbiddenDeps trips over these empty directories, so make them
trivial Go packages. They will fill with live shortly anyway.

Release note: None

tbg committed:
internal/client: git mv *.go to kv package

Release note: None

tbg committed:
internal/client: Goland-move to pkg/kv

Release note: None

tbg committed:
kv: remove placeholder file

Release note: None

tbg committed:
bulk: fix an import path

Not amending the commit since it already skews.

Release note: None

tbg committed:
github-post: fix TestListFailures

Test needed an updated package name.

Release note: None

tbg committed:
kv: fix various go generate invocations

Needed to adjust relative paths.

Release note: None

tbg committed:
cloud,kv,kvcoord: fix lint

Release note: None
